src/entry.rs

```diff
@@ -56,9 +60,47 @@ impl Entry {
         let entry = Self::new(start_hash, *cur_hashes, transactions);
         *start_hash = entry.id;
         *cur_hashes = 0;
+        assert!(serialized_size(&entry).unwrap() <= BLOB_DATA_SIZE as u64);
         entry
     }
 
+    /// Creates the next entries for given transactions, outputs
+    /// updates start_hash to id of last Entry, sets cur_hashes to 0
+    pub fn from_transactions_mut(
+        start_hash: &mut Hash,
+        cur_hashes: &mut u64,
+        transactions: Vec<Transaction>,
+    ) -> Vec<Self> {
+        let mut chunk_len = transactions.len();
+
+        // check for fit, make sure they can be serialized
+        while serialized_size(&Entry {
+            num_hashes: 0,
+            id: Hash::default(),
+            transactions: transactions[0..chunk_len].to_vec(),
+        }).unwrap() > BLOB_DATA_SIZE as u64
+        {
+            chunk_len /= 2;
+        }
+
+        let mut entries = Vec::with_capacity((transactions.len() / chunk_len) + 1);
+        for chunk in transactions.chunks(chunk_len) {
+            entries.push(Self::new_mut(start_hash, cur_hashes, chunk.to_vec()));
```

Comment:
I think there should be a `*cur_hashes = 1;` right after this line, and that `new_mut()` should have asserted on `cur_hashes` being zero when `chunk.to_vec().len()` was non-zero.